src/randomInt.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Generate an inclusive random integer between `from` and `to`.
+ *
+ * @param from - The minimum value.
+ * @param to - The maximum value.
+ * @returns The random integer.
+ * @signature
+ *   R.randomInt(from, to)
+ * @example
+ *   R.randomInt(1, 10) // => 5
+ *   R.randomInt(1n, 10n) // => 7n
+ * @dataFirst
+ * @category Number
+ */
+export function randomInt<T extends bigint | number>(
+  from: T,
+  to: Widen<T>,
+): Widen<T> {
+  if (to <= from) {
+    throw new RangeError(`to(${to}) should be greater than from(${from})`);
+  }
+
+  if (typeof from === "bigint" && typeof to === "bigint") {
+    return BigInt(
+      Math.floor(Number(to - from + BigInt(1)) * Math.random() + Number(from)),
+    ) as never;
+  }
+
+  if (!Number.isFinite(from)) {
+    throw new TypeError(`from(${from}) is not a finite number`);
+  }
+
+  if (!Number.isFinite(to)) {
+    throw new TypeError(`to(${to}) is not a finite number`);
+  }
```

Comment:
1. These should be RangeError's, as they consider the range of values; they aren't the wrong type (infinity is a number).
2. As above, add the function name to the message.
3. I'm not sure this is worthwhile, as they add a check at runtime for *everyone* just to catch a really obscure edge case that would result in a semi-erroneous (`Infinity`) value that the user can handle downstream instead.
```suggestion
```

Reply:
Removed